concord_cli/kill_task.py

```diff
@@ -1,49 +1,184 @@
+        return iterable[index-1]
+    return map(verify_index, indicies)
+
+def prompt_selection(iterable, heading, printer):
+    """ Takes 'iterable' and displays it in an ascii table using printable
+    method to determine how to transform the data. Will also prompt user for
+    a selection and return the blob to the caller"""
+    def preprinter(value_pair):
+        idx, data = value_pair
+        return [str(idx)] + printer(data)
+
+    # Create table
+    heading.insert(0, 'Idx')
+    table_data = zip(xrange(1, len(iterable) + 1), iterable)
+    table_data = map(preprinter, table_data)
+    table_data.insert(0, heading)
+
+    # Print table in nice format, wait for user selection
+    print AsciiTable(table_data).table
+    inpt = ""
+    while inpt == "":
+        inpt = raw_input('Selection: [1..%d, 1,x,y,..., quit(q), all(a)]:'\
+                         ' ' % len(iterable))
+
+    return parse_input(iterable, inpt)
+
+def prompt_nodes(pcomp_metadata):
+    def pcomp_printable(pcomp_metadata):
+        return [pcomp_metadata.taskId,
+                str(pcomp_metadata.cpus),
+                str(pcomp_metadata.mem)]
+
+    print '\nSelect a node to terminate:'
+    nodedetail_header = ['Task id', 'cpus', 'mem']
+    selection = prompt_selection(pcomp_metadata, nodedetail_header, pcomp_printable)
+    return map(lambda c: c.taskId, selection)
+
+def prompt_computations(pcomp_layout):
+    def layout_printable(pcomp_layout):
+        def pretty_md(stream_metadata):
+            return "\n".join(map(lambda c: '(%s <-> %s)' % (c.name, c.grouping),
+                                 stream_metadata))
+        return [pcomp_layout.name,
+                pretty_md(pcomp_layout.istreams),
+                pretty_md(pcomp_layout.ostreams)]
+
+    print '\nSelect a computation to inspect:'
+    header = ['Computation name', 'istreams/grouping', 'ostreams/grouping']
+    selection = prompt_selection(pcomp_layout, header, layout_printable)
+    return flatten(map(lambda c: c.nodes, selection))
+
+def interactive_mode(zookeeper, zk_path):
+    """ presents to the user an easy to use prompt so that they may selectively
+    search through computations and eventually choose a concord node to kill"""
+    print 'Querying zookeeper for cluster topology...'
+    meta = get_zookeeper_metadata(zookeeper)
+
+    # Exit on failure, or if no computations exist
+    if meta == None:
+        logger.critical('Could not connect to zk')
+        return
+    elif meta.computations == None or len(meta.computations) == 0:
+        logger.info('No computations to report')
+        return
+
+    selection = meta.computations.values()
+    user_path = [prompt_computations, prompt_nodes]
+    user_state = 0
+    while user_state < len(user_path):
+        current_action = user_path[user_state]
+        selection = current_action(selection)
+        user_state = user_state + 1
+
+    # Terminate selected node
+    kill(zookeeper, zk_path, *selection)
```

Comment:
As it stands the user can enter all at any time. It aggregates all info and sends to the next section. So in the computations view if you select all, you will see all nodes in the next pane. Select all again and every computation gets killed. Or you can use the -a flag without using interactive mode